src/Morphic-Widgets-PolyTabs/TabSelectorMorph.class.st

```diff
@@ -31,6 +31,16 @@ TabSelectorMorph >> addTab: aStringOrMorph [
 	self updateTabs
 ]
 
+{ #category : #adding }
+TabSelectorMorph >> addTab: aStringOrMorph beforeIndex: index [
+
+	self tabs add: (self newLabelMorph: aStringOrMorph) beforeIndex: index.
+	selectedIndex >= index ifTrue: [ 
+		"there is now new tab before selected tab"
+		self selectedIndex: selectedIndex + 1 ]. 
```

Comment:
This was a missing part. If we do not shift selection index then widget will think that selection is not changed and will not update inserted tab in case when it will be seleted in subsequent calls.